js/toolbar/z-plane-list.js

```diff
@@ -188,14 +266,34 @@ function ZPlaneList(parent) {
         width = width ? width : _self._zPlaneContainer.offsetWidth - 70;
         var totalWidthOfSingleIndex = _self._thumbnailProperties.width + 2 * (_self._carouselStyle.padding + _self._carouselStyle.margin + 2);
         var pageSize = Math.floor(parseFloat(width)/totalWidthOfSingleIndex);
-        if (_self.pageSize == null || pageSize < _self.pageSize) {
-            _self.pageSize = pageSize;
-            _self.collection = _self.collection.splice(0, pageSize);
-            _self._renderZPlaneCarousel();
-        } else if (pageSize > _self.pageSize) {
+
+        // in case the new page size is equal to the current page size, return
+        if (_self.pageSize != null && pageSize == _self.pageSize) {
+            return;
+        }
+
+        if (_self.currentZPlaneRequest.type != 'none') {
```

Comment:
Cleanup the `if-else` here so it's not multi-level (the `if`s that you have under `else` can be just `else if` for this `if`). Also comment all of them (only one case is commented right now)

Reply:
Done